src/xkit.cs

```diff
@@ -0,0 +1,1144 @@
+#elif MONOMAC
+		CGGlyph GetCGGlyph (nuint glyphIndex, out bool isValidIndex);
+#else
+		CGGlyph GetGlyph (nuint glyphIndex, ref bool isValidIndex);
+#endif
+
+		[Mac (10,11)]
+		[Export ("CGGlyphAtIndex:")]
+#if XAMCORE_4_0
+		CGGlyph GetGlyph (nuint glyphIndex);
+#elif MONOMAC
+		CGGlyph GetCGGlyph (nuint glyphIndex);
+#else
+		CGGlyph GetGlyph (nuint glyphIndex);
+#endif
+
+		[Mac (10,11)]
+		[Export ("processEditingForTextStorage:edited:range:changeInLength:invalidatedRange:")]
+#if XAMCORE_4_0
+		void ProcessEditing (NSTextStorage textStorage, NSTextStorageEditActions editMask, NSRange newCharacterRange, /* NSInteger */ nint delta, NSRange invalidatedCharacterRange);
+#else
+		void ProcessEditing (NSTextStorage textStorage, NSTextStorageEditActions editMask, NSRange newCharRange, /* NSInteger */ nint delta, NSRange invalidatedCharRange);
+#endif
+
+		// This method can only be called from
+		// NSLayoutManagerDelegate.ShouldGenerateGlyphs, and that method takes
+		// the same IntPtr arguments as this one. This means that creating a
+		// version of this method with nice(r) types (arrays instead of
+		// IntPtr) is useless, since what the caller has is IntPtrs (from the
+		// ShouldGenerateGlyphs parameters). We can revisit this if we ever
+		// fix the generator to have support for C-style arrays.
+		[Mac (10,11)]
+		[Export ("setGlyphs:properties:characterIndexes:font:forGlyphRange:")]
+#if XAMCORE_4_0
+		void SetGlyphs (IntPtr glyphs, IntPtr properties, IntPtr characterIndexes, NSFont font, NSRange glyphRange);
+#else
+		void SetGlyphs (IntPtr glyphs, IntPtr props, IntPtr charIndexes, NSFont aFont, NSRange glyphRange);
+#endif
+
+#if !(XAMCORE_4_0 || MONOMAC)
+		[Sealed]
+#endif
+		[Mac (10,11)]
+		[Export ("truncatedGlyphRangeInLineFragmentForGlyphAtIndex:")]
+		NSRange GetTruncatedGlyphRangeInLineFragment (nuint glyphIndex);
+
+#if !(XAMCORE_4_0 || MONOMAC)
+		[Obsolete ("Use 'GetTruncatedGlyphRangeInLineFragment' instead.")]
+		[Mac (10,11)]
+		[Export ("truncatedGlyphRangeInLineFragmentForGlyphAtIndex:")]
+		NSRange TruncatedGlyphRangeInLineFragmentForGlyphAtIndex (nuint glyphIndex);
+#endif
+
+		[Mac (10,11)]
+		[Export ("enumerateLineFragmentsForGlyphRange:usingBlock:")]
+		void EnumerateLineFragments (NSRange glyphRange, NSTextLayoutEnumerateLineFragments callback);
+
+		[Mac (10,11)]
+		[Export ("enumerateEnclosingRectsForGlyphRange:withinSelectedGlyphRange:inTextContainer:usingBlock:")]
+		void EnumerateEnclosingRects (NSRange glyphRange, NSRange selectedRange, NSTextContainer textContainer, NSTextLayoutEnumerateEnclosingRects callback);
+
+		[Protected] // Can be overridden
+		[Export ("showCGGlyphs:positions:count:font:matrix:attributes:inContext:")]
+		void ShowGlyphs (IntPtr glyphs, IntPtr positions, nuint glyphCount, NSFont font, CGAffineTransform textMatrix, NSDictionary attributes, [NullAllowed] CGContext graphicsContext);
+
+		[Protected] // "You should never call this method, but you might override it."
```

Comment:
Add comment (or similar) to an `[Advice]` attribute.

Reply:
fwiw it's the `IntPtr` that made me look into the API... but in this case I can see value in avoiding the conversion cost

Reply:
This is a C-based array, so we generator doesn't support binding it nicely.

Also if the only point to the API is to be able to override it, there's no need for a nicer API that can only be called, not overridden.